### ISSUE TITLE
cannot use firebase/analytics with Tag Manager

### ISSUE THREAD

**Issue by FelschR** [firebase/firebase-js-sdk#2338]
### My environment
  * Operating System version: NixOS 20.03pre200231.7827d3f4497
  * Browser version: Chromium 78.0.3904.87
  * Firebase SDK version: 7.2.3
  * Firebase Product: analytics

### The problem
When Google Tag Manager is already setup on the page the firebase/analytics SDK doesn't work.
I wasn't able to find any documentation on running the firebase/analytics SDK and Tag Manager together.
The only thing I found was the "*Google Analytics: App + Web Configuration*" tag to integrate Firebase Analytics via the Tag Manager instead. But it appears one can't use the firebase/analytics SDK with that setup.

#### Steps to reproduce:
- add Google Tag Manager scripts into website.
- add Firebase SDK including analytics to website via npm import
- load the website and see that there are no Analytics or Tag Manager requests sent
- remove the GTM scripts
- reload the page and see that Analytics requests are now being sent

#### Relevant Code:
My GTM setup:
```html
<head>
    ...
    <script>
        (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({"gtm.start":
        new Date().getTime(),event:"gtm.js"});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!="dataLayer"?"&l="+l:"";j.async=true;j.src=
        "https://www.googletagmanager.com/gtm.js?id="+i+dl;f.parentNode.insertBefore(j,f);
        })(window,document,"script","dataLayer","GTM-XXXXXXX");
    </script>
</head>
<body>
    <noscript>
        <iframe src="https://www.googletagmanager.com/ns.html?id=GTM-XXXXXXX"
            height="0" width="0" style="display:none;visibility:hidden"></iframe>
    </noscript>
    ...
</body>
```

My firebase/analytics SDK setup:
```javascript
import * as firebase from "firebase/app";
import "firebase/analytics";

const app = firebase.initializeApp({
    ...
});
const analytics = app.analytics();
```

**Comment by hsubox76** [comment]
Seems like the problem is we have it set up to work for existing `gtag.js` users but did not test for GTM users. Firebase analytics requires the `gtag` script, so we insert the tag and download it for you if it's not already included in the page.  We detect if it exists on the page by checking for the existence of an accompanying `window.dataLayer`.  GTM gives a false positive here because it does have a `window.dataLayer` but doesn't download `gtag.js`.

I'm working on a fix to detect this case correctly and load the `gtag` script.  In the meantime, a quick workaround would be to add a script tag to the page that downloads the gtag script:

`<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXX&l=dataLayer"></script>`

The ID here would be your Firebase `measurementId` (found in your Firebase config snippet), not your GTM container ID.

**Comment by hsubox76** [comment]
This should now be fixed by https://github.com/firebase/firebase-js-sdk/pull/2339 which was released in Firebase 7.4.0.
